corehq/apps/reports/standard/monitoring.py

```diff
@@ -1461,8 +1470,18 @@ def get_admins_and_demo_users(self):
     @property
     def users_to_iterate(self):
         if toggles.EMWF_WORKER_ACTIVITY_REPORT.enabled(self.request.domain):
-            user_query = EMWF.user_es_query(self.domain, self.request.GET.getlist(EMWF.slug))
-            return util.get_simplified_users(user_query)
+            user_query = LocationRestrictedMobileWorkerFilter.user_es_query(
+                self.domain, self.request.GET.getlist(LocationRestrictedMobileWorkerFilter.slug)
+            )
+            users = util.get_simplified_users(user_query)
+            if not self.request.couch_user.has_permission(self.domain, 'access_all_locations'):
+                accessible_location_ids = (SQLLocation.active_objects.accessible_location_ids(
+                    self.request.domain,
+                    self.request.couch_user)
+                )
+                accessible_user_ids = set(user_ids_at_locations(accessible_location_ids))
+                users = [u for u in users if u.user_id in accessible_user_ids]
```

Comment:
`user_query` above is an unevaluated ESQuery instance, so you could chain this filter logic in there, something like:
```python
if not has_permission:
  user_query = user_query.filter(accessible_location_ids)
```
which would let ES handle the filtering